### PR TITLE
HV:common:fix "integer type violations"

### PR DESCRIPTION
**Pull request by HuiHuangShi**
fix integer type violations,keep some violations which
related to hypcall and msix_entry_index.

V1->V2:1.modified API_MAJOR_VERSION from Makefile
       2.sync acrn_common.h changed to device model

Signed-off-by: Huihuang Shi <huihuang.shi@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

File: devicemodel/include/public/acrn_common.h [hunk 1]
```diff
@@ -286,7 +286,7 @@ struct acrn_vm_pci_msix_remap {
  * layout below 1M, DM add this address to E280 reserved range to make sure
  * there is no overlap for the address 0xef000 usage.
  */
-#define GUEST_CFG_OFFSET	0xef000
+#define GUEST_CFG_OFFSET	0xd0000U
```
Comment (dongyaozu) [inline devicemodel/include/public/acrn_common.h]: WHy you change it?

File: hypervisor/include/public/acrn_common.h [hunk 2]
```diff
@@ -266,7 +266,7 @@ struct acrn_vm_pci_msix_remap {
  * layout below 1M, DM add this address to E280 reserved range to make sure
  * there is no overlap for the address 0xef000 usage.
  */
-#define GUEST_CFG_OFFSET	0xef000
+#define GUEST_CFG_OFFSET	0xd0000UL
```
Comment (dongyaozu) [inline hypervisor/include/public/acrn_common.h]: same here

**Comment by HuiHuangShi** [comment]
I will review again after every rebase to guarantee quality.

**Comment by wenlingz** [comment]
verify+1

**Pull request merged**